test/stubs/Config.ts

```diff
@@ -0,0 +1,7 @@
+import sinon from "sinon";
+import { Config } from "../../src/extension/config";
+
+const stubConfig: Config = sinon.stub(Config) as any;
```

Comment:
[C] You could use 
```
const stubConfig = sinon.createStubInstance(Config) as Config;
```
here instead.

Reply:
generally I would create a new one of these for each test because I get worried (and am paranoid) about mutation leading to unexpected results but what you have works and that would be a big change.

Reply:
yah that makes sense, I can easily move this into each test now as a dependency to remove the global nature.